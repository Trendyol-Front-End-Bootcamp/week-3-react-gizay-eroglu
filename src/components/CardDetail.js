function CardDetail({
  name,
  status,
  species,
  gender,
  image,
  locationName,
  lastFiveEpisodes = [],
}) {
  return (
    <div className='card-detailed'>
      <img src={image} alt='Character' />
      <div className='card-text'>
        <h5>Name: {name}</h5>
        <span>Status: {status}</span>
        <span>Species: {species}</span>
        <span>Gender: {gender}</span>
        <span>Location: {locationName}</span>
        <h5>Last 5 Episodes:</h5>
        <div className='episode-heading'>
          <div>
            <b>Name of the Episode</b>
          </div>
          <div>
            <b>Air Date of the Episode</b>
          </div>
          <div>
            <b>Episode</b>
          </div>
        </div>
        <div className='episodes'>
          {lastFiveEpisodes.map((episode, index) => {
            return (
              <div className='episode' key={index}>
                <div>{episode.name}</div>
                <div>{episode.air_date}</div>
                <div>{episode.episode}</div>
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
}

export default CardDetail;
