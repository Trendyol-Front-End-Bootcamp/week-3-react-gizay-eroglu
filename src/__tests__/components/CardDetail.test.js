import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { shallow, configure } from "enzyme";

import CardDetail from "../../components/CardDetail";

configure({adapter: new Adapter()});

describe("Card detail component test", function () {

  let wrapper;

  beforeEach(() => { 
    wrapper = shallow(<CardDetail />);
  })

  it("should render cardDetail", () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  it("should check class of loading element which includes character detail card", () => {
    const classOfDiv = wrapper.hasClass("card-detailed");
    expect(classOfDiv).toEqual(true);
  })

  it("should detail card include image alt inside of it", () => {
      const imgAlt = wrapper.find("img").prop("alt");
      expect(imgAlt).toEqual("Character");
  })

  it("should have div which includes card texts", () => {
      const cardTextDiv = wrapper.find(".card-text")
      expect(cardTextDiv).toBeTruthy();
  })

  it("should episodes return episode list", () => {
    const episodeList = [ { name: "first episode" }, { name: "second episode" } ];
    const component = shallow (<CardDetail lastFiveEpisodes={episodeList} />);
    expect(component.findWhere( n => n.type() === "div" && n.contains("first episode")));
  })
});
