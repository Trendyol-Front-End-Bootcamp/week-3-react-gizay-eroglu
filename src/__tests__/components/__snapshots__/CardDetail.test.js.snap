// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card detail component test should render card detail 1`] = `ShallowWrapper {}`;
